docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+# Service Catalog API
```

Comment:
@duglin agree on the glossary section - I'll create one.

I think "Service Controller API" doesn't fully encapsulate what this document represents, since the service controller is a single part of the entire service catalog system. Also, it matches the name of the SIG itself, fwiw